### DISCUSSION TITLE
Change in Signature in [@types/react]`FunctionComponent` is making upgrading a lot more work

### DISCUSSION THREAD

**Discussion by mildfuzz** [DefinitelyTyped/DefinitelyTyped#66820] [Issues with a @types package]
in `@types/react@16.14.5`, the `FunctionComponent` type is as follows:  
```  
  interface FunctionComponent<P = {}> {  
(props: PropsWithChildren<P>, context?: any): ReactElement<any, any> | null;  
propTypes?: WeakValidationMap<P>;  
contextTypes?: ValidationMap<any>;  
defaultProps?: Partial<P>;  
displayName?: string;  
}  
```

where as in `@latest` is looks like 
```
interface FunctionComponent<P = {}> {
        (props: P, context?: any): ReactElement<any, any> | null;
        propTypes?: WeakValidationMap<P> | undefined;
        contextTypes?: ValidationMap<any> | undefined;
        defaultProps?: Partial<P> | undefined;
        displayName?: string | undefined;
    }
```

Because `P` not longer defined as `PropsWithChildren<P>`, it means any component that uses `FunctionComponent` no longer has an defined `children` prop, meaning that I get type errors for every component defined with this type that has children.

This means a lot of work, across libraries and a large webapp, to update all the types to include `children` explicitly.

Is there a way around this?

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
The general recommendation is to update libraries to their latest versions since most of them fixed compat issues (or we fixed it for them). For compat issues in code owned by you, you should check out https://github.com/eps1lon/types-react-codemod#implicit-children.

But keep in mind that we don't recommend this automatic addition of children: https://solverfox.dev/writing/no-implicit-children
